src/PSDMatrices.jl

```diff
@@ -42,6 +42,8 @@ function logdet(M::PSDMatrix)
     return 2 * logdet(M.R)
 end
 
+diag(M::PSDMatrix) = sum(abs2, M.R, dims=1)
```

Comment:
I think the tests might fail because `sum` returns a 1xN matrix.

Something like
```suggestion
diag(M::PSDMatrix) = Vector(sum(abs2, M.R, dims=1)[1, :])
```
might do the trick? There might also be a better solution.

Reply:
```julia
diag(M::PSDMatrix) = sum(abs2, M.R, dims=1)[:]
```
would also work, but both essentially copy the whole thing into a fresh vector :/ 

I came up with this
```julia
function diag(M::PSDMatrix)
    out = similar(M.R, size(R, 2))
    sum!(abs2, out', M.R)
    return out
end
```
It's not a one-liner anymore, but well.. Or we just use the loop ;) 

Here's also another benchmark
```julia
f1(R) = sum(abs2, R, dims=1)[:]
f2!(out, R) = (sum!(abs2, out', R); out)
f2(R) = f2!(similar(R, size(R, 2)), R)

d = 2; R = rand(d, d); out = zeros(d);
@btime f1(R);
  # 88.073 ns (2 allocations: 160 bytes)
@btime f2(R);
  # 47.950 ns (1 allocation: 80 bytes)
@btime f2!(out, R);
  # 25.886 ns (0 allocations: 0 bytes)

d = 2000; R = rand(d, d); out = zeros(d);
@btime f1(R);
  # 2.438 ms (2 allocations: 31.50 KiB)
@btime f2(R);
  # 2.437 ms (1 allocation: 15.75 KiB)
@btime f2!(out, R);
  # 2.441 ms (0 allocations: 0 bytes)
```

Reply:
> I came up with this

I think that's a good solution. This `f2/f2!` is also what I had in mind when I mentioned `sum!` in one of the comments.

Reply:
Yeah it just took me a while to figure out how to sum over the correct axis :sweat_smile: